### PR TITLE
build: upgrade to go1.17

### LINKED ISSUE #250: Upgrade to go1.17
1. Run `go mod tidy -go=1.17` to upgrade the contents of `go.mod`
   * If this creates a 2nd `require` block, combine the two and re-run the tidy command to re-sort the entries
2. Run `make fmt` to inject any needed `go:build` directives 
3. Make sure everything still builds & tests properly locally
4. Update CircleCI to use 1.17

### PR DESCRIPTION
**Pull request by danxmoran**
Closes #250

### CODE REVIEW

File: pkg/signals/context_test.go [hunk 1]
```diff
@@ -1,4 +1,5 @@
-//+build !windows
+//go:build !windows
```
Comment (davidby-influx) [inline pkg/signals/context_test.go:1]: Do we want to move entirely to the new directive now, or wait until `+build` is deprecated?

File: scripts/ci/install-go.sh [hunk 2]
```diff
@@ -1,25 +1,47 @@
 #!/usr/bin/env bash
 set -eo pipefail
 
-declare -r GO_VERSION=1.16.5
+declare -r GO_VERSION=1.17
+
+# Hashes are from the table at https://golang.org/dl/
```
Comment (davidby-influx) [inline scripts/ci/install-go.sh:6]: Any easy way to automate pulling the hashes directly? Or too much trouble...

**Review by davidby-influx — APPROVED** [review]
Non-blocking questions for my own edification only.

File: scripts/ci/install-go.sh (hunk 2 above)
Comment (danxmoran) [reply scripts/ci/install-go.sh:6]: I think we'd need to parse the HTML of the page, so probably too much trouble :) having them hard-coded here is (IMO) part of the point anyways, so we don't have to worry about having our request to pull the hashes getting corrupted / intercepted.

File: pkg/signals/context_test.go (hunk 1 above)
Comment (danxmoran) [reply pkg/signals/context_test.go:1]: I'll see if `gofmt` allows for just the new directive, my impression from reading the release notes was that it'd enforce having both styles 👀

Comment (danxmoran) [reply pkg/signals/context_test.go:1]: Didn't even get as far as running the build... GoLand immediately flags the `//go:build` with a "//go:build comment without //+build comment" error if I delete the `// +build` line

**Pull request merged**